velox/docs/functions/spark/binary.rst

```diff
@@ -13,3 +13,7 @@ Binary Functions
 .. spark:function:: xxhash64(x) -> integer
 
     Computes the xxhash64 of x.
+
+.. spark:function:: sha1(x) -> string
```

Comment:
In Spark, this function appears to allow both VARCHAR and VARBINARY input: https://docs.databricks.com/sql/language-manual/functions/sha1.html

This implementation allows only VARBINARY. Should we add support for VARCHAR as well?

Reply:
Thank you for the review!


In [SparkSQL code](https://github.com/apache/spark/blob/e6a84fef2fa43b880c89f2f54a5fa78033021fb1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/hash.scala#L172), the input type of the sha1 function expression  is binary. Spark implicitly inserts a cast expression when the sha1 input column is String type. So I think we are fine with just supporting VARBINARY.

```
select sha1(a) from testtb2;

== Physical Plan ==
* Project (3)
+- * ColumnarToRow (2)
   +- Scan parquet default.testtb2 (1)

(1) Scan parquet default.testtb2
Output [1]: [a#0]
ReadSchema: struct<a:string>

(3) Project [codegen id : 1]
Output [1]: [sha1(cast(a#0 as binary)) AS sha1(a)#4]
Input [1]: [a#0]
```

Reply:
@izchen Thank you for explaining.